### ISSUE TITLE
 Calculated loss differs from logged loss in training_step (even if seed_everything, deterministic set to true and shuffle to false)

### ISSUE THREAD

**Issue by Sulfyderz** [Lightning-AI/pytorch-lightning#19748]
### Bug description

Hello,

After training a model, I calculate the loss for the train set. Unfortunately, it doesn't match the one logged in MLFlow with the MLFlowLogger. I set `seed_everything(42)`, `deterministic=True` for the trainer and `shuffle=False` for the trainloader.
For the validation set, however it matches perfectly. Did I miss something?

I'm not sure if it's related to the issue #5247.

### What version are you seeing the problem on?

v2.2

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by Sulfyderz** [comment]
Hello,

After a new test, it works fine. My bad !

Best,
